backend/django/core/views/api_annotate.py

```diff
@@ -437,20 +442,67 @@ def data_unlabeled_table(request, project_pk):
     for d in unlabeled_data:
         serialized_data = DataSerializer(d, many=False).data
 
-        sorted_similarity_pairs = []
-        for pair in serialized_data["similarityPair"]:
-            sorted_similarity_pairs.append(pair)
         temp = {
             "Text": serialized_data["text"],
             "metadata": serialized_data["metadata"],
-            "similarityPair": serialized_data["similarityPair"],
             "ID": d.id,
         }
         data.append(temp)
 
     return Response({"data": data})
 
 
+@api_view(["POST"])
+@permission_classes([AllowAny])
+def embeddings_calculations(request):
+    """This calculates embeddings for a given array of strings.
+
+    Args:
+        strings: The array of strings
+        request: The POST request
+    Returns:
+        data: a list of data information
+    """
+
+    embeddings = model.encode(request.data["strings"])
+
+    return Response(embeddings)
+
+
+@api_view(["GET"])
+@permission_classes((IsAdminOrCreator,))
+def embeddings_comparison(request, project_pk):
+    """This finds the highest scoring labels when comparing cosine similarity scores of
+    their embeddingsfor a given input string.
+
+    Args:
+        text: The input string
+        request: The GET request
+        project_pk: Primary key of the project
+    Returns:
+        data: a list of data information
+    """
+    project = Project.objects.get(pk=project_pk)
+    project_labels = Label.objects.filter(project=project)
+    project_labels_embeddings = list(
+        LabelEmbeddings.objects.filter(label_id__in=project_labels).values_list(
+            "embedding", flat=True
+        )
+    )
+
+    text = request.GET.get("text")
+    text_embedding = model.encode(text)
+
+    cosine_scores = util.pytorch_cos_sim(text_embedding, project_labels_embeddings)
+    values, indices = cosine_scores[0].topk(5)
+
+    suggestions = []
+    for index in indices:
+        suggestions.append(LabelSerializer(project_labels[int(index)]).data)
```

Comment:
I believe you can pass serializers a list of objects with the argument `many=True`  instead of looping as you do here

Reply:
You can, but do I want to in this case? I only want to access the 5 at the different indices

Reply:
Lol that's a matter of opinion really. It might be *slightly* faster to use list comprehension like `suggestions = LabelSerializer([project_labels[int(index)]).data for index in indices],many=True)`, but given that the length is always 5 it might not really matter. I'll leave it up to you.